src/inputs/AddressInput/index.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { ReactElement, useState, ChangeEvent, useEffect } from 'react';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import TextFieldMui from '@material-ui/core/TextField';
+import styled from 'styled-components';
+
+import { trimSpaces } from '../../utils/strings';
+import {
+  addNetworkPrefix,
+  checksumAddress,
+  getAddressWithoutNetworkPrefix,
+  getNetworkPrefix,
+  isChecksumAddress,
+  isValidAddress,
+  isValidEnsName,
+} from '../../utils/address';
+import { throttle } from '../../utils/throttle';
+
+type AddressInputProps = {
+  id?: string;
+  name: string;
+  address: string;
+  networkPrefix?: string;
+  showNetworkPrefix?: boolean;
+  label: string;
+  placeholder?: string;
+  defaultValue?: string;
+  disabled?: boolean;
+  required?: boolean;
+  onChangeAddress: (address: string) => void;
+  getAddressFromDomain?: (name: string) => Promise<string>;
+  showLoadingSpinner?: boolean;
+  inputAdornment?: ReactElement | undefined | false;
+  error?: string | undefined;
+  helperText?: string | undefined;
+};
+
+function AddressInput({
+  id,
+  name,
+  address,
+  networkPrefix,
+  showNetworkPrefix,
+  label,
+  placeholder,
+  defaultValue,
+  disabled,
+  onChangeAddress,
+  getAddressFromDomain,
+  inputAdornment,
+  showLoadingSpinner,
+  error = '',
+  helperText,
+  ...rest
+}: AddressInputProps): React.ReactElement {
+  const [prefix, setPrefix] = useState(showNetworkPrefix ? networkPrefix : '');
+  const [isLoadingENSResolution, setIsLoadingENSResolution] = useState(false);
+
+  function onChange(e: ChangeEvent<HTMLInputElement>) {
```

Comment:
Mixed use of arrow functions and classic functions :wink:

Reply:
Done!